### DISCUSSION TITLE
Hanariu - Templ Atomic Design System

### DISCUSSION THREAD

**Discussion by Riu** [a-h/templ#748] [Show and tell]
Hello, 

I would like to share with you information about a project I am creating using Templ. It is called ~~Gopress~~ Hanariu and it is a concept of creating an atomic design system based on Templ, together with Tailwind, Alpine.js and HTMX.

The idea seemed crazy at first, but after some time it turned out that these could be absolutely fantastic solutions, for example for e-commerce.

I will be very grateful for your feedback. The project address is:

~~https://gopress.io~~
https://hanariu.dev

Best regards

**Comment by a-h** [comment]
Wow, this looks really impressive. Great site, and love that it's using all the same tools!

templ is designed to be able to be used to build component libraries, so I'm glad it's working out.

It looks like it's closed source at the moment. What's the plan? Open source with commercial licensing?

**Reply by Riu** [reply]
Thank you! I am so glad to hear this from you. 

We will probably want to make the code available under the MIT license (as I mentioned on the page - this decision will significantly impact the future development of the project). For now, we are validating a few things, preparing the remaining components for the first release (mostly e-commerce) and looking for partners for this project who would like to use it in commercial implementations.

**Reply by a-h** [reply]
Wow, MIT licensed?, that would be _really_ cool!

I'm speaking at https://bigskydevcon.com/ in a few weeks, and would be happy to give it a shout out if it's GA by then.

**Reply by Riu** [reply]
There's not much time until June 8, but we can try to deliver as much as we can. I can give you update next week(Monday).

**Reply by Riu** [reply]
@a-h we are officially announcing that there will be an MIT license - [FAQ](https://gopress.io/faq), but the first release will be as we planned earlier - in August. So, if you would like to mention us at the conference, great and we will be grateful, but if not - nothing happens :)

Would you like to preview the repositories before the first release?

**Reply by a-h** [reply]
I'd love to, thanks!

**Reply by a-h** [reply]
By the way, I wondered if you were interested in the Storybook support that's in https://github.com/a-h/templ/blob/main/storybook/storybook.go

You might find the example that's in there interesting. Could be a good way for people to try out your components in an interactive playground.

I'm not sure if it works any more, there was a bug introduced in Storybook which I think is now fixed. I might dust off support for Storybook again, and more properly support it if there's interest.

**Reply by Riu** [reply]
Thx! I will check it!

**Comment by DaanVervacke** [comment]
Amazing work, looking forward to the release!

**Comment by Riu** [comment]
We would like move repositories to @gopress organization but someone reserved this name for organization. ~~I'm not sure when~~ _Joined GitHub on November 25, 2014_. There is no way to contact with owner of that namespace.

**Reply by alejandrojnm** [reply]
put it under @gopressio maybe work

**Reply by alejandrojnm** [reply]
Any Idea when will be public ??

**Reply by Riu** [reply]
@alejandrojnm we are thinking about namespace/organization. First release will be in August.

**Reply by joerdav** [reply]
@gopresso would mean your branding is easy ;) "Take a shot of gopresso"

**Reply by indaco** [reply]
I would go for a different name at all 😃 

 thinking to what exists in the JS/TS ecosystems we have VitePress, SveltePress etc. 

All are for easily create static websites (documentation etc.)

For sure, I'm not aware of the overall vision for your org, but based on what you shared so far (design system, ui components library etc) I would definitely go for a complete different name. 

My 2 cents

**Reply by Riu** [reply]
So after some consideration - it will be a completely different name. Details will be announced soon on our Discord and right after that here too.

**Comment by amoghyermalkar123** [comment]
@Riu this looks fabulous! Enrolled for waitlist. Glad to see a fresh breath of tech in this space.

**Comment by Riu** [comment]
I'm happy to announce that the code for the first components of Hanariu (previously Gopress) - now available on Github. To find out more:
- check website [https://hanariu.dev](https://t.co/KjIQ25R4lX)
- add star to repo - [https://github.com/hanariu/hanariu](https://t.co/j2QQsbqrzP)
- watch this video: - [Hanariu - Templ Atomic Design System](https://www.youtube.com/watch?v=XuiXPvOD-XQ)

**Reply by joerdav** [reply]
@Oudwins FYI

**Reply by Oudwins** [reply]
@joerdav thanks! Will add this to the list
